pkgs/firehose/lib/src/changelog.dart

```diff
@@ -4,69 +4,64 @@
 
 import 'dart:io';
 
-import 'package:collection/collection.dart';
-
 class Changelog {
+  static const _headerLinePrefix = '## ';
+
   final File file;
 
   Changelog(this.file);
 
   bool get exists => file.existsSync();
 
+  static final _versionRegex = RegExp(r'\d+\.\d+\.\d+(?:[\-+]\w+)?');
+
   String? get latestVersion {
     var input = latestHeading;
 
-    if (input == null) {
-      return null;
+    if (input != null) {
+      var match = _versionRegex.firstMatch(input);
+      if (match != null) {
+        var version = match[0];
+        return version;
+      }
     }
 
-    final versionRegex = RegExp(r'[0-9]+\.[0-9]+\.[0-9]+([-\+]\w+)?');
-
-    var match = versionRegex.firstMatch(input);
-
-    if (match != null) {
-      var version = match.group(0);
-      return version;
-    }
     return null;
   }
 
   String? get latestHeading {
     var sections = _parseSections();
-    // Remove all leading "#"
-    return sections.firstOrNull?.title.replaceAll(RegExp(r'^#*'), '').trim();
```

Comment:
The only lines we accept below as title lines start with `## `, so it's overkille to match any number of `#`s here.